BuddySave.IntegrationTests/CloudManagerTests.cs

```diff
@@ -0,0 +1,287 @@
+using BuddySave.TestTools;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace BuddySave.IntegrationTests
+{
+    public class CloudManagerTests
+    {
+        [Theory, AutoMoqData]
+        public async Task UploadSave_DeletesOldCloudSave_WhenCloudSaveExists(
+            string oldSaveFileName,
+            GameSave gameSave,
+            CloudManager sut)
+        {
+            // Arrange
+            using var cloudSave = new TempSaveDir(gameSave.CloudPath);
+            using var localSave = new TempSaveDir(gameSave.LocalPath);
+            gameSave.CloudPath = cloudSave.Path;
+            gameSave.LocalPath = localSave.Path;
+
+            cloudSave.Create();
+            localSave.Create();
+
+            var oldSavePath = Path.Combine(cloudSave.Path, oldSaveFileName);
+            await File.WriteAllTextAsync(oldSavePath, "Test");
+
+            // Act
+            sut.UploadSave(gameSave);
+
+            // Assert
+            Assert.False(Directory.Exists(oldSavePath));
+        }
+
+        [Theory, AutoMoqData]
+        public void UploadSave_CopiesEmptyLocalToCloud_WhenCloudSaveDoesNotExist(
```

Comment:
I don't get it, what does it mean `CopiesEmptyLocal`?

Reply:
This is kinda a stepping stone - copy only an empty folder. Later tests checks the files.

I will rename to CopySaveFolderToCloud.